node/src/components/contract_runtime/operations.rs

```diff
@@ -305,13 +310,19 @@ pub fn execute_finalized_block(
             ProofHandling::NoProofs,
         ));
 
+        let category = transaction.transaction_kind();
+
```

Comment:
this variable should be called `transaction_kind`

the currently shadowed variable post execution can retain the name `category`